hibernate-java8/src/main/java/org/hibernate/type/ZonedDateTimeType.java

```diff
@@ -53,7 +54,7 @@ public ZonedDateTime next(ZonedDateTime current, SessionImplementor session) {
 	@Override
 	@SuppressWarnings("unchecked")
 	public Comparator<ZonedDateTime> getComparator() {
-		return ComparableComparator.INSTANCE;
+		return ZonedDateTimeComparator.INSTANCE;
```

Comment:
For some reason the JDK does not offer a "timeline comparator" for ZonedDateTime like they do for OffsetDateTime, so we'd definitely need your ZonedDateTimeComparator.